src/TwoFactorAuthenticationPlugin.php

```diff
@@ -47,21 +50,44 @@ public function register(Panel $panel): void
                 ]);
         }
 
+        $panel
+            ->authMiddleware([
+                $this->challengeTwoFactorCheckMiddleware,
+            ]);
+
         if ($this->hasEnforcedTwoFactorSetup()) {
             $panel
                 ->authMiddleware([
-                    EnforceTwoFactorSetup::class,
+                    $this->enforceTwoFactorSetupMiddleware,
                 ]);
         }
     }
 
-    public function enforceTwoFactorSetup(bool $condition = true): static
+    public function setChallengeTwoFactorMiddleware(string $middleware = ChallengeTwoFactor::class): static
```

Comment:
Can we rename the `setChallengeTwoFactorMiddleware ` method to `setTwoFactorChallengeMiddleware`